obstacle_stop_planner_refine/test/src/test_obstacle_stop_planner.cpp

```diff
@@ -36,19 +64,179 @@ class ObstacleStopPlannerTest : public ::testing::Test
+
+  ASSERT_FALSE(candidate_obstacles.has_value());
+}
+
+TEST_F(ObstacleStopPlannerTest, findCollisionPoint_has_value)
+{
+  std::vector<Point3d> points;
+  points.emplace_back(0.0, 0.0, 0.0);
+  points.emplace_back(10.0, 0.0, 0.0);
+  points.emplace_back(20.0, 0.0, 0.0);
+  points.emplace_back(30.0, 0.0, 0.0);
+  points.emplace_back(40.0, 0.0, 0.0);
+  points.emplace_back(50.0, 0.0, 0.0);
+  auto trajectory = convertPointsToTrajectoryWithYaw2(points);
+
+  std::vector<Point3d> obstacles;
+  obstacles.emplace_back(30.1, 0.0, 0.0);
+  obstacles.emplace_back(30.2, 0.0, 0.0);
+  obstacles.emplace_back(30.3, 0.0, 0.0);
+
+  const auto candidate_obstacles = planner_->findCollisionPoint(trajectory, obstacles);
+
+  ASSERT_TRUE(candidate_obstacles.has_value());
+  ASSERT_EQ(candidate_obstacles.value().obstacle_point, obstacles.at(0).to_2d());
+}
+
+TEST_F(ObstacleStopPlannerTest, planSlowDown)
+{
+  std::vector<Point3d> points;
+  points.emplace_back(0.0, 0.0, 0.0);
+  points.emplace_back(10.0, 0.0, 0.0);
+  points.emplace_back(20.0, 0.0, 0.0);
+  points.emplace_back(30.0, 0.0, 0.0);
+  points.emplace_back(40.0, 0.0, 0.0);
+  points.emplace_back(50.0, 0.0, 0.0);
+  auto trajectory = convertPointsToTrajectoryWithYaw2(points);
+
+  // Add velocity
+  for (size_t i = 0; i < trajectory.points.size(); ++i) {
+    trajectory.points.at(i).twist.linear.x = static_cast<double>(i);
+  }
+
+  // Set parameter
+  slow_down_param_->enable_slow_down = true;
+  slow_down_param_->min_slow_down_vel = 2.0;
+  slow_down_param_->max_slow_down_vel = 3.0;
+  slow_down_param_->max_deceleration = 1.0;
+  slow_down_param_->expand_slow_down_range = 1.0;
+
+  planner_->updateParameters(stop_param_, slow_down_param_, acc_param_);
+
+  obstacle_stop_planner::Collision collision;
+  collision.segment_index = 3;
+  collision.obstacle_point = Point2d{35.0, 0.0};
+
+  const auto processed_trajectory = planner_->planSlowDown(trajectory, collision);
+
+  // Get expected velocity
+  const auto lateral_deviation = autoware_utils::calcLateralDeviation(trajectory.points.at(3).pose, autoware_utils::toMsg(collision.obstacle_point.to_3d()));
+
+  const auto target_velocity = planner_->calcSlowDownTargetVel(lateral_deviation);
+
+  ASSERT_EQ(processed_trajectory.points.at(0).twist.linear.x, 0.0);
+  ASSERT_EQ(processed_trajectory.points.at(1).twist.linear.x, 1.0);
+  ASSERT_EQ(processed_trajectory.points.at(2).twist.linear.x, 2.0);
+  ASSERT_EQ(processed_trajectory.points.at(3).twist.linear.x, target_velocity);
+  ASSERT_EQ(processed_trajectory.points.at(4).twist.linear.x, target_velocity);
+  ASSERT_EQ(processed_trajectory.points.at(5).twist.linear.x, target_velocity);
+}
+
+TEST_F(ObstacleStopPlannerTest, planObstacleStop)
+{
+  std::vector<Point3d> points;
+  points.emplace_back(0.0, 0.0, 0.0);
+  points.emplace_back(10.0, 0.0, 0.0);
+  points.emplace_back(20.0, 0.0, 0.0);
+  points.emplace_back(30.0, 0.0, 0.0);
+  points.emplace_back(40.0, 0.0, 0.0);
+  points.emplace_back(50.0, 0.0, 0.0);
```

Comment:
Or you can write like this? Then it can be const

```cpp
const std::vector<Point3d> points{
  {0.0, 0.0, 0.0},
  {10.0, 0.0, 0.0},
  {20.0, 0.0, 0.0},
  {30.0, 0.0, 0.0},
  {40.0, 0.0, 0.0},
  {50.0, 0.0, 0.0},
};
```